js/PaymentRequest/index.js

```diff
@@ -258,9 +271,38 @@ export default class PaymentRequest {
         INTERNAL_SHIPPING_ADDRESS_CHANGE_EVENT,
         this._handleShippingAddressChange.bind(this)
       );
+
+      this._paymentMethodChangeSubscription = DeviceEventEmitter.addListener(
+          INTERNAL_PAYMENT_METHOD_CHANGE_EVENT,
+          this._handlePaymentMethodChange.bind(this)
+      );
     }
   }
 
+  _handlePaymentMethodChange(paymentMethod: PaymentMethod) {
+    this._paymentMethod = paymentMethod.paymentMethodType;
+
+    if(this._paymentMethod === 'PKPaymentMethodTypeCredit') {
+      this._details = this._creditDebitDetails.credit;
+    }
+    else if (this._paymentMethod === 'PKPaymentMethodTypeDebit') {
+      this._details = this._creditDebitDetails.debit;
+    }
```

Comment:
This is just prototype code (nice prototype code with unit tests!), but this is one part that i'd add a todo for.  Handle the else statement.

Reply:
good point for clarification, the to-do has been added.

we have another investigation card about that the supported cards can only be shown up in apple pay flow to avoid other card problem, and this is handle in this investigation card https://trello.com/c/jXpIHzT6/2280-investigate-apm-mobile-only-show-up-the-supported-cards.